Source/Totem/Runtime/Timeline/Topic.cs

```diff
@@ -41,7 +41,8 @@ protected void ThenSchedule(Event e, TimeSpan timeOfDay)
 
 		protected void ThenSchedule(Event e, IEnumerable<TimeSpan> timesOfDay)
 		{
-			ThenSchedule(e, timesOfDay.Select(GetWhenOccursNext).Min());
+ 			Expect(timesOfDay.Any(), "Event requires at least one time for scheduling.");
```

Comment:
`.Any()` will attempt to enumerate the sequence, which is done again in the `.Select()` call. We don't want to assume the sequence can be iterated twice (a common constraint with `IEnumerable<>`).

What if we attempt to generate the sequence via `GetWhenOccursNext` and check that we have a result?

`.Min()` will throw an exception for an empty sequence; we could put `DateTime.MinValue` in there if it's empty, and then check for that as the result. If so, we know there were no times of day in the sequence.

Something like

```csharp
var result = timesOfDay.Select(GetWhenOccursNext).DefaultIfEmpty(DateTime.MinValue).Min();

Expect(result).IsNot(DateTime.MinValue, "Event requires at least one time for scheduling");

return result;
```